contract/package.json

```diff
@@ -73,13 +76,47 @@
   },
   "homepage": "https://github.com/agoric-labs/dapp-join-game#readme",
   "eslintConfig": {
+    "env": {
```

Comment:
Whatever knowledge you have learned about lint config, would you please contribute it to...

 - https://github.com/Agoric/agoric-sdk/discussions/8274

Reply:
Sure, let me gather some feedback here first to make sure the approach is correct. The main changes are - 
 1. use typescript as the eslint parser
 2. add `no-void` and `no-floating-promises` rules
 3. reimplement `_ignore` patterns for unused-vars, as `@typescript-eslint` also checks for this (curious - can we dedupe `no-unused-vars` and `@typescript-eslint/no-usused-vars`. Both are effectively doing the same thing)?
 

Here would be the updated directions from the [current answer](https://github.com/Agoric/agoric-sdk/discussions/8274#discussioncomment-6940968):
1. Run:
```diff
yarn add -D \
 eslint \
 @agoric/eslint-config@dev \
 @endo/eslint-plugin \
 @jessie.js/eslint-plugin \
  eslint-config-airbnb-base \
 eslint-plugin-jsdoc \
 eslint-config-prettier \
 eslint-plugin-import \
- eslint-plugin-github 
+ eslint-plugin-github \
+ @typescript-eslint/eslint-plugin
+ @typescript-eslint/parser
+ typescript
```

2. Add the following to your `package.json`
```diff
 "eslintConfig" : {
+  "env": {
+    "node": true
+  },
+  "parser": "@typescript-eslint/parser",
   "parserOptions": {
-      "sourceType": "module",
-      "ecmaVersion": 6
+    "project": "./tsconfig.json",
+    "sourceType": "module",
+    "ecmaVersion": 2020
   },
   "extends": [
+    "plugin:@typescript-eslint/recommended",
     "@agoric"
   ],
+  "plugins": [
+    "@typescript-eslint",
+    "prettier"
+  ],
+  "rules": {
+    "@typescript-eslint/no-floating-promises": "warn",
+    "no-void": [
+      "error",
+      {
+        "allowAsStatement": true
+      }
+    ],
+    "prettier/prettier": "warn",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      {
+        "vars": "all",
+        "args": "all",
+        "argsIgnorePattern": "^_",
+        "varsIgnorePattern": "^_"
+      }
+    ]
+  },
 }

```
3. Add a tsconfig.json file (new)
```diff
+{
+  "compilerOptions": {
+    "noEmit": true,
+    "target": "esnext",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "skipLibCheck": true,
+    "checkJs": false,
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "maxNodeModuleJsDepth": 2,
+    "strict": true
+  },
+  "include": ["**/*.js", "**/*.ts"],
+  "exclude": ["bundles"]
+}
```

4. Run `yarn eslint '**/*.{js,ts}'` (or specify other files/directory globs). Run `yarn tsc` to lint types.